reset_install_requirements-no-darcs.sh

```
@@ -22,90 +22,150 @@ check_prerequisites() {
+    git clone --depth 1 --filter=blob:none --sparse "$REPO_URL" basilisk-C-temp
 
     if [[ $? -ne 0 ]]; then
-        echo "\033[0;31mError: Failed to extract basilisk.tar.gz\033[0m"
+        printf "\033[0;31mError: Failed to clone repository\033[0m\n"
+        echo "URL: $REPO_URL"
         exit 1
     fi
 
-    # Clean up the tar file
-    rm basilisk.tar.gz
+    cd basilisk-C-temp
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)